documentation/projects/proposals/infrastructure_improvements/20240105_implementation_plan_move_remaining_services_to_openverse_org.md

```diff
@@ -0,0 +1,738 @@
+   the `generic/service-domain` instances pointing those domains to the load
+   balancer, otherwise the next root module will overwrite the new settings.
+   - After this, control over requests to `openverse.engineering` API domains
+     will rest solely with the Cloudflare redirect rule, and forwarded to the
+     appropriate `openverse.org`.
+1. Remove Cloudflare configuration for `openverse.engineering` API domains from
+   rules and Access. This will require manually deleting resources in the
+   Cloudflare dashboard, as well as changes to the `cloudflare` root module (for
+   Access).
+
+### 3. Update references, announce to Make Blog, and finalise
+
+1. Update references to `openverse.engineering` domains in our public and
+   internal documentation to use `openverse.org` instead.
+1. Publish a Make blog post announcing the change.
+1. Decomission all resources from `openverse.engineering` except the redirect
+   rule. At this point, I think we can also remove the
+   `search.openverse.engineering` settings, as no one should be using it anymore
+   and it was never a "public" access point for the Openverse frontend (we only
+   used it in the iframe and never advertised it directly).
+1. Downgrade `openverse.engineering` Cloudflare plan to the free tier.
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+### Preliminary work
+
+The first of these two steps are straightforward, with little possible
+additional description to the tasks other than what I've written above. Instead,
+I'll share the rationale for the changes here. I do so in order of complexity,
+from least to most complex. I've added specific steps for the networking module
+changes because those are more complex.
+
+#### Moving Cloudflare Access and the new `cloudflare` root module
+
+Moving the Cloudflare Access configuration out of the legacy root modules is
+just one more step in our progress towards eliminating the legacy Terraform
+configuration. There's nothing about the module that requires it to be in the
+legacy root environments, so this should be a fairly easy one to move, but will
+require importing resources into the new root module and using
+`terraform state rm` to remove them from the legacy root module, in order to
+prevent them from being deleted if the legacy root module is applied.
+
+We'll move this to a new `cloudflare` root module rather than one of the `next`
+environment root modules, because there is no specific environment related to
+these settings, and it's best to keep these all together in a single place.
+Later on, in the Airflow and API work, we'll expand the Cloudflare configuration
+in `cloudflare` to include the page and WAF rules related to these services that
```

Comment:
Noting also that I've added some Cache Rules which mirror the Page Rules in an attempt to [more accurately use Cloudflare's intended settings for caching](https://blog.cloudflare.com/introducing-cache-rules/), so those will need to be moved over as well!